wordpress-mobile/AztecEditor-iOS
# Obliterating Fragmentation: Mark I + II

### Details:
This PR also contains the changes introduced in #560:

- NSAttributedStringToNodes Internal API has been updated, so that Elements are first extracted, and Branches assembled 'by hand'
- Sort + Merge algorithm has been implemented, so that we minimize fragmentation (Left to Right analysis).
- Plus: Since i was around, added support to serialize Pre elements (+ Unit Tests!!)

### Testing:
1. Run the unit tests
2. Run a smoke test over the Out Converter
3. Add a `Pre` element, verify it gets serialized properly
4. Try snippets that should generate fragmentation. Few samples below!

```
<u><b>Bold</b>Underlined<i>Italics</i></u>
```

```
<u><i>1</i></u>
<u><b><i>2</i></b></u>
<u><b>3</b></u>
<u><b>4</b></u>
```

cc @diegoreymendez
